File: codeForces/0110A.py

def lucky(n):
	cnt = 0
	while n > 0:
		d = n % 10
		if d == 4 or d == 7:
			cnt += 1
		n = n // 10
		
	if cnt==0:
		print("NO")
		return
		
	while cnt > 0:
		d = cnt % 10
		if d != 4 and d != 7:
			print("NO")
			return
		cnt = cnt // 10
	print("YES")

n = int(input())
lucky(n)

File: CodeForces/0110A.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long

bool lucky(ll n) {
	ll cnt = 0;
	while (n > 0) {
		cnt += (n%10 == 4) || (n%10 == 7);
		n /= 10;
	}
	if (!cnt) return false;
	
	while (cnt > 0) {
		if ((cnt%10 != 4) && (cnt%10 != 7)) return false;
		cnt /= 10;
	}
	return true;
}

int main () {
	ll n;
	cin >> n;
	if (lucky(n)) {
		cout << "YES\n";
	} else {
		cout << "NO\n";
	}
	return 0;
}

// Lorenzo Fiorini
